scipy/stats/stats.py

```diff
@@ -7469,6 +7470,254 @@ def brunnermunzel(x, y, alternative="two-sided", distribution="t",
+       Available: https://www.jstor.org/stable/2957563.
+    .. [2] H. A. David and H. N. Nagaraja, "Order Statistics in
+       Nonparametric Inference" in Order Statistics,
+       John Wiley & Sons, Ltd, 2005, pp. 159-170.
+       Available: https://www.jstor.org/stable/2957563.
+    .. [3] N. Hutson, A. Hutson, L. Yan, "QuantileNPCI:
+       Nonparametric Confidence Intervals for Quantiles,"
+       R package, https://cran.r-project.org/package=QuantileNPCI
+
+
+    Examples
+    --------
+    >>> from scipy.stats import confint_quantile
+    >>> x = [2, 8, 3, 6, 4, 1, 5, 9, 7]
+    >>> confint_quantile(x, 0.5, 0.95)
+    (2, 8)
+
+    To compute a two-sided interval instead, use the `type` parameter.
+
+    >>> confint_quantile(x, 0.5, 0.99, type='two-sided')
+    (1, 9)
+
+    You can also pass the number of samples as argument (instead of the
+    samples) themselves. The returned values are then the indexes of the
+    upper and lower bounds for the confidence intervals.
+
+    >>> N = 20
+    >>> confint_quantile(N, 0.75, 0.90)
+    (11, 17)
+
+
+    .. versionadded:: 1.6.0
+    """
+
+    ##
+    # Checking the inputs
+    #
+    # x can be either an integer or a one-dimensional array-like
+    if isinstance(x, int):
+        if x < 1:
+            raise ValueError(
+                """Invalid data 'x': %s,
+            `x` must be either a strictly positive integer
+            or one-dimensional array-like.""" % repr(x))
+        n = x
+        # The function will returns the confint indexes
+        return_index = True
+    else:
+        x = np.asarray(x)
+        if x.ndim != 1:
+            raise ValueError(
+                """Invalid data 'x': %s,
+            `x` must be either a strictly positive integer
+            or one-dimensional array-like.""" % repr(x))
+        x = np.sort(x, axis=0)
+        n = x.shape[0]
+        # The function returns the confint as values of x
+        return_index = False
+    #
+    # `confidence` and `quantile` must be between 0 and 1
+    if confidence >= 1 or confidence <= 0:
+        raise ValueError(
+            """Invalid `confidence`: %s.
+            Provide a real number strictly between 0 and 1."""
+            % repr(confidence))
+    if quantile >= 1 or quantile <= 0:
+        raise ValueError(
+            """Invalid `quantile`: %s.
+            Provide a real number strictly between 0 and 1."""
+            % repr(quantile))
+    #
+    # `type` can be only `one-sided` or `two-sided`
+    if not (type == 'one-sided' or type == 'two-sided'):
+        raise ValueError(
+            """Invalid parameter: %s.
+            Valid 'type' values: 'one-sided' or 'two-sided'""" % repr(type))
+    ##
+
+    # Handle the type of intervals (one- or two-sided)
+    if type == 'two-sided':
+        conf_working = (1+confidence)/2
+    else:
+        # type == 'one-sided'
+        conf_working = confidence
+
+    # Compute the lower bound
+    LB = _confint_lowerbound(n, quantile, conf_working)
```

Comment:
Can you point me to where in the references I can check this against?

Reply:
Do you mean: where in the reference is the math explained/presented? 
- In the Thompson paper: eq. (7) on page 125
- In the David and Nagaraja book: chapter 7, eq (7.1.4), page 160